File: ryosuke/python/knock003.py
if __name__=='__main__':
    sent = "Now I need a drink, alcoholic of course, after the heavy lectures involving quantum mechanics."
    pi = list()
    for tok in sent.split():
        if tok.endswith(',') or tok.endswith('.'):
            tok = tok[:-1]
        pi.append(len(tok))
    print pi

File: ryosuke/java/knock_2015java/src/knock_2015java/Knock003.java
/**
 * 
 */
package knock_2015java;

import java.util.ArrayList;

/**
 * @author ryosuke
 *
 */
public class Knock003 extends Knock {
		String sent;
		ArrayList<Integer> pi;
		/**
	 * 
	 */
	public Knock003() {
		pi = new ArrayList<Integer>();
		sent = "Now I need a drink, alcoholic of course, after the heavy lectures involving quantum mechanics.";
	}

	/* (非 Javadoc)
	 * @see knock_2015java.Knock#start()
	 */
	@Override
	public void start() {
		for (String s : sent.split(" ")){
			s = s.replace(".", "");
			s = s.replace(",", "");
			pi.add(s.length());
		}
		for (int i : pi){
			System.out.print(i);
		}
	}

}
